lib/pacemaker/pcmk_verify.c

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2023 the Pacemaker project contributors
+ *
+ * The version control history for this file may have further details.
+ *
+ * This source code is licensed under the GNU Lesser General Public License
+ * version 2.1 or later (LGPLv2.1+) WITHOUT ANY WARRANTY.
+ */
+
+#include <crm_internal.h>
+#include <crm/cib/internal.h>
+#include <crm/common/output.h>
+#include <crm/common/results.h>
+#include <crm/common/scheduler.h>
+#include <pacemaker-internal.h>
+#include <pacemaker.h>
+
+#include <stdint.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include "libpacemaker_private.h"
+
+int
+pcmk__verify(pcmk_scheduler_t *scheduler, pcmk__output_t *out,
+                   gboolean use_live_cib, const char *xml_file,
+                   const char *xml_string, gboolean xml_stdin,
+                   const char *cib_save, int verbosity, const char* output_ty)
+{
+    int rc = pcmk_rc_ok;
+    GError *error = NULL;
+    xmlNode *cib_object = NULL;
+    xmlNode *status = NULL;
+
+    crm_info("=#=#=#=#= Getting XML =#=#=#=#=");
+
+    if (use_live_cib) {
+        crm_info("Reading XML from: live cluster");
+        rc = cib__signon_query(out, NULL, &cib_object);
+
+        if (rc != pcmk_rc_ok) {
+            // cib__signon_query() outputs any relevant error
+        }
+
+    } else if (xml_file != NULL) {
+        cib_object = filename2xml(xml_file);
+        if (cib_object == NULL) {
+            rc = ENODATA;
+            g_set_error(&error, PCMK__RC_ERROR, rc, "Couldn't parse input file: %s", xml_file);
```

Comment:
Regarding error handling throughout this function... the error message isn't going to get displayed anywhere because you're only ever setting it.  If you called `pcmk__output_and_clear_error` before exiting, that would move the error message from the GError to the output object, and then that would get displayed the right way.

But, I don't think we really want to be doing anything with GErrors here.  Typically, library functions only return an error code and it's up to the caller of the function to inspect that error code and print out something to the user.

I would instead try to come up with unique error codes for the various spots where g_set_error is being called.  If necessary, use `out->err()` to add more information.  In this particular case, where you might want to return ENODATA for no xml_file, no xml_string, no stdin, or none of the above, I think it's not worth adding extra information or printing out different error messages in the caller.  Do we really care if it's an input file or string?  The user will know what arguments they provided, so all of these could just be reported with "Couldn't parse input".